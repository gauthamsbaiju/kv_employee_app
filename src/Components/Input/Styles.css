.input-wrapper{
margin-top: 20px;
}

.label{
    font-size: 14px;
    margin-bottom: 400px;
}

.input{
    margin-top: 8px;
    margin-left: 0px;
    border-radius: 5px;
    border: 1.8px solid rgb(179, 176, 176);
    height: 48px;
    width: 100%;
}

.btn{
    background-color: #03AEEE;
    color: white;
    margin-top: 40px; 
    border-radius: 5px;
    border: 1.8px solid rgb(179, 176, 176);
    height: 48px;
    width: 100%;
}