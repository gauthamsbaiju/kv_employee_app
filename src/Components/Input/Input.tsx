import React from 'react';
import './Styles.css';

export type InputPropsType = {
  value?: string;
  onChange?: (e) => void;
  lable: string;
  placeholder: string;
  type: 'text' | 'password';
};

const Input: React.FC<InputPropsType> = (props) => {
  return (
    <div className='input-wrapper'>
      <label className='label' data-testid='label-test'>
        {props.lable}
      </label>
      <div>
        <input
          type={props.type}
          placeholder={props.placeholder}
          className='input'
          onChange={props.onChange}
          data-testid='input-test'
        ></input>
      </div>
    </div>
  );
};

export default Input;
