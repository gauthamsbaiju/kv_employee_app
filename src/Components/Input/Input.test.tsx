import { fireEvent, render, screen } from '@testing-library/react';
import Input, { InputPropsType } from './Input';
import '@testing-library/jest-dom';

describe('Test for input component', () => {
  test('Test for label rendered', () => {
    const InputProps: InputPropsType = {
      lable: 'Input',
      placeholder: 'Input',
      type: 'text'
    };

    render(<Input {...InputProps} />);
    const element = screen.getByTestId('label-test');

    if (element) expect(element).toHaveTextContent('Input');
  });

  test('Test for snapshot rendered', () => {
    const InputProps: InputPropsType = {
      lable: 'Input',
      placeholder: 'Input',
      type: 'text'
    };

    render(<Input {...InputProps} />);
    const element = screen.getByTestId('input-test');

    expect(element).toMatchSnapshot();
  });

  test('Test for input type', () => {
    const InputProps: InputPropsType = {
      lable: 'Input',
      placeholder: 'Input',
      type: 'text'
    };

    render(<Input {...InputProps} />);
    const element = screen.getByTestId('input-test');

    expect(element).toHaveAttribute('type', 'text');
  });

  test('Test for input placeholder', () => {
    const InputProps: InputPropsType = {
      lable: 'Input',
      placeholder: 'Input',
      type: 'text'
    };

    render(<Input {...InputProps} />);
    const element = screen.getByTestId('input-test');

    expect(element).toHaveAttribute('placeholder', 'Input');
  });

  test('Test for on change', () => {
    const onChange = jest.fn();
    const InputProps: InputPropsType = {
      lable: 'Input',
      placeholder: 'Input',
      type: 'text',
      onChange
    };

    render(<Input {...InputProps} />);
    const element = screen.getByTestId('input-test');

    fireEvent.change(element);
    expect(onChange).toBeCalled;
  });
});
