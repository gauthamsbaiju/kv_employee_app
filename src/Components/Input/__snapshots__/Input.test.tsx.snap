// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for input component Test for snapshot rendered 1`] = `
<input
  class="input"
  data-testid="input-test"
  placeholder="Input"
  type="text"
/>
`;
